### PR TITLE
Add API for comment responses and Paginated projects

### PR DESCRIPTION
**Pull request by bamandel**
Added paginated projects. The request builder used the wrong query params originally. I'm not sure if that would require a major version jump or not.

### CODE REVIEW

File: src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java [hunk 1]
```diff
@@ -32,8 +32,8 @@
 import com.synopsys.integration.rest.HttpUrl;
 
 public class JiraRequestFactory {
-    public static final String LIMIT_PARAMETER = "limit";
-    public static final String OFFSET_PARAMETER = "offset";
+    public static final String LIMIT_PARAMETER = "maxResults";
```
Comment (gkillough) [inline src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java:35]: Is this the equivalent of "limit" or of "total"?

**Review by gkillough — APPROVED** [review]

File: src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java (hunk 1 above)
Comment (bamandel) [reply src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java:35]: Yes, these are the values they use in their pagination. We had the wrong values originally https://developer.atlassian.com/cloud/jira/platform/rest/v2/intro/#expansion

**Review by ChomickiM — APPROVED** [review]

File: src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java (hunk 1 above)
Comment (bamandel) [reply src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java:35]: Oh sorry, to answer your question, it's the equivalent of total I believe.

File: src/main/java/com/synopsys/integration/jira/common/cloud/service/IssueService.java [hunk 2]
```diff
@@ -154,13 +155,9 @@ public TransitionsResponseModel getTransitions(String issueIdOrKey) throws Integ
...
 
-    public void addComment(IssueCommentRequestModel requestModel) throws IntegrationException {
+    public IssueCommentResponseModel addComment(IssueCommentRequestModel requestModel) throws IntegrationException {
         HttpUrl commentsUri = createApiCommentsUri(requestModel.getIssueIdOrKey());
-        JiraResponse response = jiraCloudService.post(requestModel, commentsUri);
-
-        if (response.isStatusCodeError()) {
-            throw new IntegrationException(String.format("Error commenting on issue; cause: (%d) - %s", response.getStatusCode(), response.getStatusMessage()));
-        }
+        return jiraCloudService.post(requestModel, commentsUri, IssueCommentResponseModel.class);
```
Comment (jamesrichard91) [inline src/main/java/com/synopsys/integration/jira/common/cloud/service/IssueService.java:160]: I assume the error handling is now included in the post?

File: src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java (hunk 1 above)
Comment (jamesrichard91) [reply src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java:35]: Do these apply to both Cloud and Server?

File: src/main/java/com/synopsys/integration/jira/common/cloud/service/IssueService.java (hunk 2 above)
Comment (bamandel) [reply src/main/java/com/synopsys/integration/jira/common/cloud/service/IssueService.java:160]: Correct, all our clients now throw rest exceptions when an error status is received. I believe that is a reasonable assumption for a library service. (We did that for almost all error status's originally anyway)

File: src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java (hunk 1 above)
Comment (bamandel) [reply src/main/java/com/synopsys/integration/jira/common/model/request/JiraRequestFactory.java:35]: Yes, the expansion section at the beginning here goes over the server expectations https://docs.atlassian.com/software/jira/docs/api/REST/7.6.1/

**Pull request merged**